utilities/general-helpers.metta

```diff
@@ -57,3 +57,7 @@
 (= (findAndReplace $old $new $list)
     (collapse (replace $old $new (superpose $list)))
 )
+
+
+(: until (-> (-> $a Bool) (-> $a $a) $a $a))
+(= (until $predicate $function $x) (if ($predicate $x) $x (let $x' ($function $x) (until $predicate $function $x'))))
```

Comment:
@Yagth is this implementation concerned with executing a function until a truth value is met?

Reply:
Yep. I was planning to use it like a while loop for some cases if possible